File: app/src/main/java/com/tm/blplayer/ui/fragment/VideoDescriptionFragment.java
package com.tm.blplayer.ui.fragment;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.tm.blplayer.R;
import com.tm.blplayer.bean.VideoDetailData;
import com.tm.blplayer.widget.GlideCircleTransform;
import com.zhy.view.flowlayout.FlowLayout;
import com.zhy.view.flowlayout.TagAdapter;
import com.zhy.view.flowlayout.TagFlowLayout;

import butterknife.BindView;
import butterknife.OnClick;

/**
 * @author wutongming
 * @description 视频详情页简介tab
 * @since 2017/5/18
 */

public class VideoDescriptionFragment extends BaseFragment {
    @BindView(R.id.tv_title)
    TextView tvTitle;
    @BindView(R.id.tv_play_count)
    TextView tvPlayCount;
    @BindView(R.id.tv_danmaku_count)
    TextView tvDanmakuCount;
    @BindView(R.id.tv_desc)
    TextView tvDesc;
    @BindView(R.id.tv_share)
    TextView tvShare;
    @BindView(R.id.tv_coin)
    TextView tvCoin;
    @BindView(R.id.tv_collect)
    TextView tvCollect;
    @BindView(R.id.tv_download)
    TextView tvDownload;
    @BindView(R.id.iv_avatar)
    ImageView ivAvatar;
    @BindView(R.id.tv_author)
    TextView tvAuthor;
    @BindView(R.id.tv_up_time)
    TextView tvUpTime;
    @BindView(R.id.flow_layout)
    TagFlowLayout flowLayout;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_video_desc;
    }

    @Override
    protected void initView() {

    }

    public void initData(VideoDetailData data) {
        tvTitle.setText(data.getTitle());
        tvDesc.setText(data.getDesc());
        VideoDetailData.PlayInfoBean playInfo = data.getPlay_info();
        VideoDetailData.AuthorInfoBean authorInfo = data.getAuthor_info();
        if (playInfo != null) {
            tvPlayCount.setText(String.valueOf(playInfo.getView()));
            tvDanmakuCount.setText(String.valueOf(playInfo.getDanmaku()));
            tvShare.setText(String.valueOf(playInfo.getShare()));
            tvCoin.setText(String.valueOf(playInfo.getCoin()));
            tvCollect.setText(String.valueOf(playInfo.getFavorite()));
        }
        if (authorInfo != null) {
            Glide.with(getActivity()).load(authorInfo.getCard().getFace()).transform(new GlideCircleTransform(getActivity())).into(ivAvatar);
            tvAuthor.setText(authorInfo.getCard().getName());
            tvUpTime.setText(getResources().getString(R.string.video_detail_up_time, data.getCreate_time()));
        }

        TagAdapter<String> adapter = new TagAdapter<String>(data.getTag_list()) {
            @Override
            public View getView(FlowLayout parent, int position, String s) {
                TextView inflate = (TextView) View.inflate(getActivity(), R.layout.adapter_video_tag, null);
                inflate.setText(s);
                return inflate;
            }
        };

        flowLayout.setAdapter(adapter);
    }

    @OnClick({R.id.tv_share, R.id.tv_coin, R.id.tv_collect, R.id.tv_download})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.tv_share:
                break;
            case R.id.tv_coin:
                break;
            case R.id.tv_collect:
                break;
            case R.id.tv_download:
                break;
        }
    }
}

File: app/src/main/java/com/tm/blplayer/ui/fragment/VideoDescriptionFragment.kt
package com.tm.blplayer.ui.fragment

import android.content.Intent
import android.support.v7.widget.LinearLayoutManager
import android.view.View
import android.widget.TextView
import com.bumptech.glide.Glide
import com.tm.blplayer.R
import com.tm.blplayer.base.BaseFragment
import com.tm.blplayer.bean.VideoDetailData
import com.tm.blplayer.bean.VideoItem
import com.tm.blplayer.listener.OnItemClickListener
import com.tm.blplayer.ui.activity.VideoDetailActivity
import com.tm.blplayer.ui.adapter.VideoCardAdapter
import com.tm.blplayer.utils.StringUtils
import com.tm.blplayer.utils.TimeUtils
import com.tm.blplayer.utils.constants.Constants
import com.tm.blplayer.widget.GlideCircleTransform
import com.zhy.view.flowlayout.FlowLayout
import com.zhy.view.flowlayout.TagAdapter
import kotlinx.android.synthetic.main.include_desc_bottom.*
import kotlinx.android.synthetic.main.include_desc_middle.*
import kotlinx.android.synthetic.main.include_desc_top.*

/**
 * @author wutongming
 * @description
 * @since 2017/6/14
 */
class VideoDescriptionFragment : BaseFragment() {
    override val layoutId: Int
        get() = R.layout.fragment_video_desc

    override fun initView() {

    }

    fun initData(data: VideoDetailData?) {
        tv_title.text = data?.title
        tv_desc.text = data?.desc
        val playInfo = data?.play_info
        val authorInfo = data?.author_info
        if (playInfo != null) {
            val viewCount = playInfo.view
            val danmakuCount = playInfo.danmaku
            val shareCount = playInfo.share
            val coinCount = playInfo.coin
            val favoriteCount = playInfo.favorite

            tv_play_count.text = if (viewCount > 10000) resources.getString(R.string.home_video_count, StringUtils.DecimalFormat(viewCount)) else viewCount.toString()
            tv_danmaku_count.text = if (danmakuCount > 10000) resources.getString(R.string.home_video_count, StringUtils.DecimalFormat(danmakuCount)) else danmakuCount.toString()
            tv_share.text = if (shareCount > 10000) resources.getString(R.string.home_video_count, StringUtils.DecimalFormat(shareCount)) else shareCount.toString()
            tv_coin.text = if (coinCount > 10000) resources.getString(R.string.home_video_count, StringUtils.DecimalFormat(coinCount)) else coinCount.toString()
            tv_collect.text = if (favoriteCount > 10000) resources.getString(R.string.home_video_count, StringUtils.DecimalFormat(favoriteCount)) else favoriteCount.toString()
        }
        if (authorInfo != null) {
            Glide.with(activity).load(authorInfo.card.face).transform(GlideCircleTransform(activity)).into(iv_avatar)
            tv_author.text = authorInfo.card.name
            tv_up_time.text = resources.getString(R.string.video_detail_up_time, TimeUtils.formatStringTime(data.create_time))
        }

        val adapter = object : TagAdapter<String>(data?.tag_list) {
            override fun getView(parent: FlowLayout, position: Int, s: String): View {
                val inflate = View.inflate(activity, R.layout.adapter_video_tag, null) as TextView
                inflate.text = s
                return inflate
            }
        }

        flow_layout.adapter = adapter

        initRecyclerView(data?.relative_list)

    }

    private fun initRecyclerView(videoItemList: List<VideoItem>?) {
        val manager = LinearLayoutManager(activity)
        rv_relative.layoutManager = manager
        val adapter = VideoCardAdapter(activity, videoItemList ?: ArrayList<VideoItem>(), false)
        adapter.setOnItemClickListener(object : OnItemClickListener<VideoItem> {
            override fun onItemClick(view: View, data: VideoItem, position: Int) {
                val intent = Intent(activity, VideoDetailActivity::class.java)
                intent.putExtra(Constants.VIDEO_AID, data.aid)
                startActivity(intent)
            }
        })
        rv_relative.adapter = adapter
        rv_relative.isNestedScrollingEnabled = false
    }

    override fun onClick(v: View?) {
        when (view?.id) {
            R.id.tv_share -> {
            }
            R.id.tv_coin -> {
            }
            R.id.tv_collect -> {
            }
            R.id.tv_download -> {
            }
        }
    }
}
